collect_app/src/test/java/org/odk/collect/android/application/initialization/SavepointsImporterTest.kt

```diff
@@ -0,0 +1,305 @@
+        savepointsImporter.run()
+
+        // verify import
+        val savepoints = projectDependencyProvider.savepointsRepository.getAll()
+        assertThat(savepoints.isEmpty(), equalTo(true))
+    }
+
+    @Test
+    fun ifABlankFormHasASavepointCreatedEarlierThanTheForm_nothingShouldBeImported() {
+        // create blank forms
+        createBlankForm(project, "sampleForm", "1", date = System.currentTimeMillis() + TimeInMs.ONE_HOUR)
+
+        // create savepoints
+        createFileInCache(project, "sampleForm_2024-04-10_01-35-41.xml.save")
+
+        // trigger importing
+        savepointsImporter.run()
+
+        // verify import
+        val savepoints = projectDependencyProvider.savepointsRepository.getAll()
+        assertThat(savepoints.isEmpty(), equalTo(true))
+    }
+
+    @Test
+    fun ifABlankFormHasASavepointButTheFormIsSoftDeleted_nothingShouldBeImported() {
+        // create blank forms
+        createBlankForm(project, "sampleForm", "1", deleted = true)
+
+        // create savepoints
+        createFileInCache(project, "sampleForm_2024-04-10_01-35-41.xml.save")
+
+        // trigger importing
+        savepointsImporter.run()
+
+        // verify import
+        val savepoints = projectDependencyProvider.savepointsRepository.getAll()
+        assertThat(savepoints.isEmpty(), equalTo(true))
+    }
+
+    @Test
+    fun ifInCacheThereIsAFileThatLooksLikeASavepointForBlankFormButDoesNotContainTheCorrectSuffixThatIdentifiesSavepoints_nothingShouldBeImported() {
+        val formName = "sampleForm"
+
+        // create blank forms
+        createBlankForm(project, formName, "1", "1")
+
+        // create savepoints
+        createFileInCache(project, "${formName}_2024-04-10_01-35-41.xml")
+
+        // trigger importing
+        savepointsImporter.run()
+
+        // verify import
+        val savepoints = projectDependencyProvider.savepointsRepository.getAll()
+        assertThat(savepoints.isEmpty(), equalTo(true))
+    }
+
+    @Test
+    fun ifThereAreMultipleDifferentBlankFormsWithSavepoints_allSavepointsShouldBeImported() {
+        val form1Name = "sampleForm1"
+        val form2Name = "sampleForm2"
+
+        // create blank forms
+        val blankForm1 = createBlankForm(project, form1Name, "1")
+        val blankForm2 = createBlankForm(project, form2Name, "2")
+
+        // create savepoints
+        val savepointFile1 = createFileInCache(project, "${form1Name}_2024-04-10_01-35-41.xml.save")
+        val savepointFile2 = createFileInCache(project, "${form2Name}_2024-04-10_01-35-42.xml.save")
+
+        // trigger importing
+        savepointsImporter.run()
+
+        // verify import
+        val savepoints = projectDependencyProvider.savepointsRepository.getAll()
+        val expectedSavepoint1 =
+            Savepoint(blankForm1.dbId, null, savepointFile1.absolutePath, "${projectDependencyProvider.storagePathProvider.getOdkDirPath(StorageSubdirectory.INSTANCES, project.uuid)}/$form1Name/$form1Name.xml")
```

Comment:
Pulling `StoragePathProvider`, `SavePointsRepository` etc out to their own fields would reduce a lot of the noise in these tests.

Reply:
Done.